src/Interpreters/Session.cpp

```diff
@@ -307,14 +309,22 @@ void Session::authenticate(const Credentials & credentials_, const Poco::Net::So
     if ((address == Poco::Net::SocketAddress{}) && (prepared_client_info->interface == ClientInfo::Interface::LOCAL))
         address = Poco::Net::SocketAddress{"127.0.0.1", 0};
 
-    LOG_DEBUG(log, "{} Authenticating user '{}' from {}",
-            toString(auth_id), credentials_.getUserName(), address.toString());
+    LOG_DEBUG(log, "Authenticating user '{}' from {}",
+            credentials_.getUserName(), address.toString());
 
     try
     {
         user_id = global_context->getAccessControl().authenticate(credentials_, address.host());
-        LOG_DEBUG(log, "{} Authenticated with global context as user {}",
-                toString(auth_id), toString(*user_id));
+        LOG_DEBUG(log, "Authenticated with global context as user {}",
+                toString(*user_id));
+
+        if (!extra_granted_roles_.empty() && global_context->getSettingsRef().allow_extenral_roles_in_interserver_queries)
```

Comment:
Probably it's better to check this setting before sending those external roles (and don't send them if it's false).

Reply:
yep, please see https://github.com/ClickHouse/ClickHouse/pull/42537/files#diff-a05838cce3aacc32a5b2fc017af198fb3086277d8912159aa7a3dcac7aeb7478R260